### DISCUSSION TITLE
v8.0.0-alpha.0

### DISCUSSION THREAD

**Discussion by grabbou** [react-native-community/cli#1567] [Q&A]
<hr /><em>This discussion was created from the release <a href='https://github.com/react-native-community/cli/releases/tag/v8.0.0-alpha.0'>v8.0.0-alpha.0</a>.</em>

**Comment by numandev1** [comment]
@grabbou what do you think about this PR https://github.com/react-native-community/cli/pull/1556 for custom placeholder, if you test it than i will write test cases for approve this

**Comment by Ferossgp** [comment]
The auto-linking of iOS native libs for some packages is broken. and it looks like the ones that overrides ios `dependency` in the [config](https://github.com/facebook/flipper/blob/main/react-native/react-native-flipper/react-native.config.js) or [config](https://github.com/bugsnag/bugsnag-js/blob/next/packages/react-native/react-native.config.js).

Does not reproduce with `7.0.2`

Also, I can see this warning:
`warn Package react-native-flipper has been ignored because it contains invalid configuration. Reason: "dependency.platforms.ios.project" is not allowed`

**Reply by thymikee** [reply]
cc @grabbou (he's on vacation right now, so might respond later)

**Reply by kyle-ssg** [reply]
Just an FYI for people in case they come across this, this is why some plugins (e.g. redux-flipper) have stopped working and for the message 
```
The native module for Flipper seems unavailable. Please verify that `react-native-flipper` is installed as yarn dependency to your project and, for iOS, that `pod install` is run in the `ios` directory.
```

It looks like the following patch works for iOS and Android (you'll need to rebuild/pod install)

```
diff --git a/node_modules/react-native-flipper/react-native.config.js b/node_modules/react-native-flipper/react-native.config.js
index 9b962ed..35cbe80 100644
--- a/node_modules/react-native-flipper/react-native.config.js
+++ b/node_modules/react-native-flipper/react-native.config.js
@@ -8,13 +8,4 @@
  */
 
 module.exports = {
-  dependency: {
-    platforms: {
-      ios: {
-        // This file should not be needed at all, but without it.
-        // cli 4.2.2 will search for an xcode project
-        project: '.',
-      },
-    },
-  },
 };

```

**Comment by mateusz1913** [comment]
👋  I noticed that with v8 there are less project configuration properties for Android project ([old](https://github.com/react-native-community/cli/blob/7.x/docs/projects.md#project-1) vs [new](https://github.com/react-native-community/cli/blob/master/docs/projects.md#project-1)). In new version there is no possibility to specify custom path for build and settings Gradle scripts.

My use case would be to have a support for Gradle scripts in Kotlin DSL (instead of Groovy DSL).
With cli versions < 8, it was possible to specify path that points to `build.gradle.kts` (and not `build.gradle`) file. Without that manual workaround, autolinking fails with `Couldn't locate file at '<path-to-app-project>/build.gradle' `.

Will it still be possible to pass different path to the configuration? Or will cli support Gradle Kotlin DSL at some point?

**Reply by thymikee** [reply]
Sounds like it makes sense for the CLI to try load `build.gradle` and if it's not there, try `build.gradle.kts`? Would you like to contribute such a feature?

**Reply by mateusz1913** [reply]
Sure, actually I tested v8 and it properly handles Kotlin DSL 👍 . I will make PRs for v6 and v7

**Reply by mateusz1913** [reply]
v6 PR - https://github.com/react-native-community/cli/pull/1578
v7 PR - https://github.com/react-native-community/cli/pull/1579

cc @thymikee

**Reply by thymikee** [reply]
Hm, since it's working with v8 as you tested and it also works with v7 with a small configuration change, maybe it's not worth to change it at all?

**Reply by mateusz1913** [reply]
I think that small configuration change is a workaround. If we can fix it, then I think it's worth it.
Also v6 & v7 will still be used for some time and as I understand, we don't need to specify exact version of cli when bootstrapping projects, so it should automatically pick latest one, shouldn't it?

**Reply by thymikee** [reply]
Yup, as long as it's not locked with a lockfile
